packages/node/src/integrations/context.ts

```diff
@@ -380,3 +393,72 @@ async function getLinuxInfo(): Promise<OsContext> {
 
   return linuxInfo;
 }
+
+/**
+ * Grabs some information about hosting provider based on best effort.
+ */
+function getCloudResourceContext(): CloudResourceContext | undefined {
+  if (process.env.VERCEL) {
+    // https://vercel.com/docs/concepts/projects/environment-variables/system-environment-variables#system-environment-variables
+    return {
+      'cloud.provider': 'vercel',
+      'cloud.region': process.env.VERCEL_REGION,
+    };
+  } else if (process.env.AWS_REGION) {
+    // https://docs.aws.amazon.com/lambda/latest/dg/configuration-envvars.html
+    return {
+      'cloud.provider': 'aws',
+      'cloud.region': process.env.AWS_REGION,
+      'cloud.platform': process.env.AWS_EXECUTION_ENV,
+    };
+  } else if (process.env.GCP_PROJECT) {
+    // https://cloud.google.com/composer/docs/how-to/managing/environment-variables#reserved_variables
+    return {
+      'cloud.provider': 'gcp',
+    };
+  } else if (process.env.ALIYUN_REGION_ID) {
+    // TODO: find where I found these environment variables - at least gc.github.com returns something
```

Comment:
Note that I haven't figured this out yet. This is not a leftover in this PR but an actual todo.